huxley/core/models.py

```diff
@@ -361,6 +361,20 @@ def remove(assignment_data):
 
         return failed_assignments
 
+    @classmethod
+    def update_assignment(cls, **kwargs):
+        assignment = kwargs['instance']
+        # Check that this assignment already exists
+        if assignment.id:
```

Comment:
nit: I'd return early to prevent too much indentation with an `if not assignment.id` instead.